packages/php-wasm/universal/src/lib/php-request-handler.ts

```diff
@@ -370,3 +368,27 @@ function seemsLikeADirectoryRoot(path: string) {
 	const lastSegment = path.split('/').pop();
 	return !lastSegment!.includes('.');
 }
+
+/**
+ * Applies the given rewrite rules to the given path.
+ *
+ * @param  path  The path to apply the rules to.
+ * @param  rules The rules to apply.
+ * @returns The path with the rules applied.
+ */
+export function applyRewriteRules(path: string, rules: RewriteRule[]): string {
+	for (const rule of rules) {
+		if (new RegExp(rule.match).test(path)) {
+			if (rule.keep) {
+				path = path.replace(
+					path.replace(new RegExp(rule.keep), ''),
+					''
+				);
+			}
+			if (rule.remove) {
+				path = path.replace(new RegExp(rule.remove), '');
+			}
```

Comment:
Is this code path actually used, or can it be removed?

```suggestion
			if (rule.remove) {
				path = path.replace(new RegExp(rule.remove), '');
			}
```

Reply:
We can remove it. I used it initially and wanted to leave it here in case it's useful in the future.